crates/sui-core/src/narwhal_manager/mod.rs

```diff
@@ -140,13 +153,13 @@ impl NarwhalManager {
             now.elapsed().as_secs_f64()
         );
 
-        *running = Running::True(committee.epoch());
+        *worker_running = Running::True(committee.epoch());
```

Comment:
Practically the only case where we won't reach this point is either when:
* we failed to start the `primary` node - where we `panic` - so we'll never proceed to start the workers
* we failed to start the `worker` node(s) - where we do also `panic` - where the process will exit

since we panic in both cases , then practically it's like running behind a single mutex. Given that - and unless I am missing something here - splitting the mutexes might not give us much more value?

Reply:
That is a good point, let's sync on this one